packages/contracts/src/utils/GsnTypes.sol

```diff
@@ -9,6 +9,7 @@ interface GsnTypes {
         uint256 gasPrice;
         uint256 pctRelayFee;
         uint256 baseRelayFee;
```

Comment:
Now that we have all the gas-denominated components of the transaction cost, there is no need to modify the actual charge calculations to solve the `externalGasLimit - gasleft()` problem. It can be done as a separate PR if needed.